packages/ember-auto-import/ts/analyzer-syntax.ts

```diff
@@ -0,0 +1,249 @@
+import { ReadStream } from 'fs';
+
+export interface LiteralImportSyntax {
+  isDynamic: boolean;
+  specifier: string;
+}
+
+export interface TemplateImportSyntax {
+  isDynamic: boolean;
+  // these are the string parts of the template literal. The first one always
+  // comes before the first expression.
+  cookedQuasis: string[];
+  // for each of the expressions in between the cookedQuasis, this is an
+  // optional hint for what to name the expression that goes there. It's
+  // optional because in general there may not be an obvious name, but in
+  // practice there often is, and we can aid debuggability by using names that
+  // match the original code.
+  expressionNameHints: (string | null)[];
+}
+
+export type ImportSyntax = LiteralImportSyntax | TemplateImportSyntax;
+
+const marker = 'eaimeta';
+
+export function serialize(imports: ImportSyntax[]): string {
+  let tokens = [];
+  for (let imp of imports) {
+    if ('specifier' in imp) {
+      tokens.push(imp.isDynamic ? 1 : 0);
+      tokens.push(imp.specifier);
+    } else {
+      tokens.push(imp.isDynamic ? 3 : 2);
+      tokens.push(imp.cookedQuasis);
+      tokens.push(imp.expressionNameHints);
+    }
+  }
+  return `${marker}${JSON.stringify(tokens).slice(1, -1)}${marker}`;
```

Comment:
How large does this stringified structure get? If it is large, could we gzip it?

Reply:
It's smaller than the import statements it is summarizing, so I think not worth zipping.

Reply:
@ef4 awsome!